runtime/gc_base/modronapi.cpp

```diff
@@ -1123,7 +1125,11 @@ j9gc_reinitialize_for_restore(J9VMThread *vmThread)
 	MM_GCExtensionsBase *extensions = env->getExtensions();
 	MM_VerboseManagerBase *verboseGCManager = extensions->verboseGCManager;
 
+	/* The checkpoint thread must release VM access for the duration of reinitializeGCThreadCountForRestore.
+	 * The thread startup/attach may require exclusive access if an af/heap OOM is encountered.  */
```

Comment:
2 things:
- simply state it may trigger GC (what is very similar to AF, but not really to heap OOM) - it's common knowledge that GC will need exclusive VM access
- need to state it'ss tartup of a new thread (rather then something about this thread triggering GC - in which case releasing VM access for this thread would be completely wrong).

So, something like this but feel free to change it further.

The checkpoint thread must release VM access for the duration of reinitializeGCThreadCountForRestore, since new GC threads could be started and the startup/attach of a new GC thread involves allocation and may trigger GC